### PR TITLE
General: refactor code to use latest libcoap

### PR DESCRIPTION
**Pull request by mrdeep1**
Aids using simplified code without the applicaion having to have too much knowledge about CoAP.

### CODE REVIEW

File: client.cc [hunk 1]
```diff
@@ -11,22 +11,48 @@
 
 static int have_response = 0;
 
+#ifndef COAP_CLIENT_URI
+#define COAP_CLIENT_URI "coap://coap.me/hello"
```
Comment (rjmrohit94) [inline client.cc:24]: Try changing the URI to "coap://coap.me/hello/xyz"

File: server.cc [hunk 2]
```diff
@@ -12,45 +12,66 @@
 int
 main(void) {
   coap_context_t  *ctx = nullptr;
-  coap_address_t dst;
   coap_resource_t *resource = nullptr;
-  coap_endpoint_t *endpoint = nullptr;
   int result = EXIT_FAILURE;;
   coap_str_const_t *ruri = coap_make_str_const("hello");
```
Comment (rjmrohit94) [inline server.cc:30]: and URI here to "hello/xyz"

File: client.cc (hunk 1 above)
Comment (mrdeep1) [reply client.cc:24]: The server `coap.me` does not have a resource defined for `/hello/xyz`, and so will return a "`4.04 not found`".

File: server.cc (hunk 2 above)
Comment (mrdeep1) [reply server.cc:30]: No plans to change the code to this.

**Comment by mrdeep1** [comment]
@rjmrohit94 Just pushed a change that now supports an optional argument to ./client where the CoAP URI can be specified instead of using the default `coap://coap.me/hello`.

**Pull request merged**